Bug fix: Default value if the environment variable is null or blank

Izzzu opened the pull request:
## What this PR changes/adds

ℹ️   This PR is a follow up for https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1064 which contained incomplete fix - changed the `if (value != null)` into `if (!StringUtils.isNullOrBlank(value))` only for system properties and not for environment variables. 

This PR modifies the _propOrEnv_ method in _ConfigurationFunctions_ class to use default value if **environment variable** is null or blank instead of checking only if it's null.

## Why it does that

Environment variable shouldn't be blank, but rather unset if that's the case. 

## Further notes

This change should fix the failing Cosmos DB integration tests on main branch.
It implements the 3rd possible solution from bug: #1057 

Added unit tests for the _ConfigurationFunctions_ class after PR feedback.

## Linked Issue(s)

Closes #1057 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

ndr-brt reviewed (commented):
I propose to add some unit tests on this method to better describe the added checks.

Izzzu commented:
> I propose to add some unit tests on this method to better describe the added checks.

@ndr-brt Actually this check was not added, just changed from `value != null` to `!StringUtils.isNullOrBlank(value)'. 
But indeed this class is not covered with tests, so as I touched this code already I added tests as you proposed. Please take a look
